ancpbids/plugins/plugin_dsloader.py

```diff
@@ -64,12 +64,26 @@ def _handle_metadata_files(self, folder):
             return
         for file in filter(lambda f: f.name.endswith(".json"), folder.files):
             if isinstance(file, Artifact):
-                mdfile = MetadataArtifact()
+                if self.options.lazy_loading:
+                    mdfile = LazyMetadataArtifact()
+                    mdfile.set_lazy_loading(True)
+                else:
+                    mdfile = MetadataArtifact()
             else:
-                mdfile = MetadataFile()
+                if self.options.lazy_loading:
+                    mdfile = LazyMetadataFile()
+                    mdfile.set_lazy_loading(True)
+                else:
+                    mdfile = MetadataFile()
+
             mdfile.parent_object_ = folder
             mdfile.update(file)
-            mdfile.contents = mdfile.load_contents()
+
+            # Load contents based on lazy loading setting
```

Comment:
This is the most crucial change: if lazy-loading chosen, then let contents be a lambda which resolves itself if the user accesses the contents property. Need further elaboration.